In [18]:
from getschools import lookup_exact, lookup_contains, lookup_both, add_exact_matches
import pandas as pd
import numpy as np

In [19]:
df = pd.read_csv("../raw/FTEStaffing-precleaned-with-OO.csv")

In [3]:
# cleanup 

df["District"] = df["District"].fillna(method="ffill")
df["School"] = df["School"].fillna(method="ffill")
df.columns = [u'district', u'school', u'category', u'type',
       u'2010-11', u'2011-12', u'2012-13', u'2013-14', u'2014-15',
       u'Unnamed: 9']

df = df[[u'district', u'school', u'category', u'type',
       u'2010-11', u'2011-12', u'2012-13', u'2013-14', u'2014-15']]

df.head()

district  \
0  Achievement First Hartford Academy Inc. District   
1  Achievement First Hartford Academy Inc. District   
2  Achievement First Hartford Academy Inc. District   
3  Achievement First Hartford Academy Inc. District   
4  Achievement First Hartford Academy Inc. District   

                                    school  \
0  Achievement First Hartford Academy Inc.   
1  Achievement First Hartford Academy Inc.   
2  Achievement First Hartford Academy Inc.   
3  Achievement First Hartford Academy Inc.   
4  Achievement First Hartford Academy Inc.   

                                            category           type 2010-11  \
0  Administrators  Coordinators and Department Ch...      Certified     6.9   
1  Counselors  Social Workers  and School Psychol...      Certified       2   
2  General Education - Paraprofessional Instructi...  Non-Certified       0   
3       General Education - Teachers and Instructors      Certified    50.6   
4                      Library/Media - Support Staff  Non-Certified       0   

   2011-12  2012-13  2013-14  2014-15  
0      7.8     12.0     15.0      1.0  
1      2.0      1.0      1.0      0.0  
2      0.0      0.0      0.0      0.0  
3     59.2     57.6     69.0      0.0  
4      0.0      0.0      0.0      0.0

In [4]:
with_codes = add_exact_matches(df)

In [5]:
with_codes.head()

district  \
0  Achievement First Hartford Academy Inc. District   
1  Achievement First Hartford Academy Inc. District   
2  Achievement First Hartford Academy Inc. District   
3  Achievement First Hartford Academy Inc. District   
4  Achievement First Hartford Academy Inc. District   

                                    school  \
0  Achievement First Hartford Academy Inc.   
1  Achievement First Hartford Academy Inc.   
2  Achievement First Hartford Academy Inc.   
3  Achievement First Hartford Academy Inc.   
4  Achievement First Hartford Academy Inc.   

                                            category           type 2010-11  \
0  Administrators  Coordinators and Department Ch...      Certified     6.9   
1  Counselors  Social Workers  and School Psychol...      Certified       2   
2  General Education - Paraprofessional Instructi...  Non-Certified       0   
3       General Education - Teachers and Instructors      Certified    50.6   
4                      Library/Media - Support Staff  Non-Certified       0   

   2011-12  2012-13  2013-14  2014-15 school_code  \
0      7.8     12.0     15.0      1.0     2880113   
1      2.0      1.0      1.0      0.0     2880113   
2      0.0      0.0      0.0      0.0     2880113   
3     59.2     57.6     69.0      0.0     2880113   
4      0.0      0.0      0.0      0.0     2880113   

                             check_sch_col  \
0  Achievement First Hartford Academy Inc.   
1  Achievement First Hartford Academy Inc.   
2  Achievement First Hartford Academy Inc.   
3  Achievement First Hartford Academy Inc.   
4  Achievement First Hartford Academy Inc.   

                                     check_dist_col  
0  Achievement First Hartford Academy Inc. District  
1  Achievement First Hartford Academy Inc. District  
2  Achievement First Hartford Academy Inc. District  
3  Achievement First Hartford Academy Inc. District  
4  Achievement First Hartford Academy Inc. District

In [6]:
len(with_codes[with_codes["school_code"].isnull()]["school"].unique())

79

In [7]:
len(with_codes[with_codes["school_code"].notnull()]["school"].unique())

1178

# Pretty good. 1178 mathes, 79 not-founds. That's still 79 headaches...

In [ ]:
for sch in with_codes[with_codes["school_code"].isnull()]["school"].unique():
    print sch

In [ ]:
for sch in with_codes[with_codes["school_code"].isnull()]["district"].unique():
    print sch

In [ ]:
lookup_contains("district","Windham School District")

In [34]:
with_codes[with_codes["school_code"].isnull()][["district","school"]].groupby(["district","school"]).agg({"school":np.size})

school
district                                           school                                                    
Area Cooperative Educational Services              Wintergreen Interdistrict Magnet School                 11
Bloomfield School District                         J. P. Vincent School                                    10
Bridgeport School District                         #NAME?                                                  20
                                                   Central at Night                                         1
                                                   New Start Program                                        2
                                                   The University School                                    2
Capitol Region Education Council                   Academy of Aerospace & Engineering - Site 2              7
                                                   Greater Hartford Academy of the Arts High Schoo...      10
                                                   Medical Professions and Teacher Preparation Aca...      10
Charter School for Young Children on Asylum Hil... Charter School for Young Children on Asylum Hil...      10
Cooperative Educational Services                   C. E. S. Special Education                               5
Coventry School District                           Coventry Early Intervention Services                     3
Danbury School District                            Mill Ridge Educational Center                            5
                                                   Other                                                    6
Derby School District                              Other                                                    1
East Hartford School District                      Saint Christopher School                                 1
                                                   Willowbrook Early Childhood                              5
Eastern Connecticut Regional Educational Servic... Arts at the Capitol Theater                              2
EdAdvance                                          GFLC/ACCESS School                                       5
Enfield School District                            Harriet Beecher Stowe School                            10
                                                   Other                                                    1
                                                   Thomas G. Alcorn School                                 10
Glastonbury School District                        Other                                                    1
Hartford School District                           Asian Studies Academy                                   11
                                                   Breakthrough II Elementary School                       10
                                                   Dr. Ramon E. Betances Early Reading Lab                 11
                                                   Great Path Academy High School at MCC                   10
                                                   Hartford Transitional Learning Academy (Secondary)       2
                                                   High School Inc.                                        10
                                                   Montessori Magnet School at Annie Fisher                11
...                                                                                                       ...
Stafford School District                           Saint Edward School                                      1
Stonington School District                         Special Needs Preschool Program                          2
Stratford School District                          Center School                                            1
                                                   Other                                                    3
Thompson School District         

In [37]:
map(lambda x: x["school"], lookup_exact("district","Waterbury School District"))

[u'Excel III',
 u'Duggan CBL',
 u'Woodrow Wilson School',
 u'Woodrow Wilson BDLC',
 u'Wilby High School',
 u'Wilby ESC',
 u'West Side SCOPE',
 u'West Side Middle School',
 u'West Side Bilingual',
 u'West Side BDLC',
 u'Wendell L. Cross School',
 u'Waterbury School District',
 u'Waterbury Career Academy',
 u'Waterbury Arts Magnet School (Middle)',
 u'Waterbury Arts Magnet School (High)',
 u'Waterbury Adult Education',
 u'Washington School',
 u'Walsh School',
 u'Wallace SCOPE',
 u'Wallace Bilingual',
 u'Wallace BDLC',
 u'WATERBURY/LV',
 u'State Street School',
 u'Sprague School',
 u'Special Education Preschool',
 u'Rotella Interdistrict Magnet School',
 u'Regan School',
 u'Reed School',
 u'Reed ESC',
 u'North End SCOPE',
 u'North End Middle School',
 u'North End ESC',
 u'North End Bilingual',
 u'North End BDLC',
 u'Michael F. Wallace Middle School',
 u'Michael F. Wallace Autism',
 u'Margaret M. Generali Elementary School',
 u'Margaret M. Generali Autism',
 u'Maloney Prek',
 u'Maloney Int

In [33]:
lookup_exact("school","Woodstock Elementary School",
             rset=lookup_exact("district","Windham School District"))

[{u'city': u'Woodstock',
  u'district': u'Woodstock School District',
  u'district_code': u'1690011',
  u'education_program': u'',
  u'fax': u'',
  u'grade_1': u'1',
  u'grade_10': u'0',
  u'grade_11': u'0',
  u'grade_12': u'0',
  u'grade_2': u'1',
  u'grade_3': u'1',
  u'grade_4': u'1',
  u'grade_5': u'0',
  u'grade_6': u'0',
  u'grade_7': u'0',
  u'grade_8': u'0',
  u'grade_9': u'0',
  u'inter_district_magnet': u'',
  u'kindergarten': u'1',
  u'phone': u'860-928-0471',
  u'pre_kindergarten': u'1',
  u'program_type': u'',
  u'school': u'Woodstock Elementary School',
  u'school_code': u'1690211',
  u'state': u'CT',
  u'street': u'24 Frog Pond Rd.',
  u'type': u'Public Schools',
  u'website': u'www.woodstockschools.net',
  u'zip': u'06281-3009'}]

In [41]:
with_codes[(with_codes["district"].str.contains("Bridgeport"))\
          & with_codes["school_code"].isnull()]

district                 school  \
611  Bridgeport School District       Central at Night   
823  Bridgeport School District      New Start Program   
824  Bridgeport School District      New Start Program   
872  Bridgeport School District  The University School   
873  Bridgeport School District  The University School   
896  Bridgeport School District                 #NAME?   
897  Bridgeport School District                 #NAME?   
898  Bridgeport School District                 #NAME?   
899  Bridgeport School District                 #NAME?   
900  Bridgeport School District                 #NAME?   
901  Bridgeport School District                 #NAME?   
902  Bridgeport School District                 #NAME?   
903  Bridgeport School District                 #NAME?   
904  Bridgeport School District                 #NAME?   
905  Bridgeport School District                 #NAME?   
906  Bridgeport School District                 #NAME?   
907  Bridgeport School District                 #NAME?   
908  Bridgeport School District                 #NAME?   
909  Bridgeport School District                 #NAME?   
910  Bridgeport School District                 #NAME?   
911  Bridgeport School District                 #NAME?   
912  Bridgeport School District                 #NAME?   
913  Bridgeport School District                 #NAME?   
914  Bridgeport School District                 #NAME?   
915  Bridgeport School District                 #NAME?   

                                              category  \
611       General Education - Teachers and Instructors   
823       General Education - Teachers and Instructors   
824       Special Education - Teachers and Instructors   
872       General Education - Teachers and Instructors   
873       Special Education - Teachers and Instructors   
896   Engineering and Physical Sciences HS at the F...   
897  Counselors  Social Workers  and School Psychol...   
898  General Education - Paraprofessional Instructi...   
899       General Education - Teachers and Instructors   
900                      Library/Media - Support Staff   
901  Other Staff Providing Non-Instructional Servic...   
902                                      School Nurses   
903  Special Education - Paraprofessional Instructi...   
904       Special Education - Teachers and Instructors   
905     Instructional Specialists Who Support Teachers   
906   Research and Zoological Studies HS at the Fai...   
907  Counselors  Social Workers  and School Psychol...   
908  General Education - Paraprofessional Instructi...   
909       General Education - Teachers and Instructors   
910                      Library/Media - Support Staff   
911  Other Staff Providing Non-Instructional Servic...   
912                                      School Nurses   
913  Special Education - Paraprofessional Instructi...   
914       Special Education - Teachers and Instructors   
915     Instructional Specialists Who Support Teachers   

                                                  type    2010-11  2011-12  \
611                                          Certified          1      0.0   
823                                          Certified        4.5      0.0   
824                                          Certified        1.5      0.0   
872                                          Certified        4.5      9.0   
873                                          Certified        1.5      4.0   
896  Administrators  Coordinators and Department Ch...  Certified      0.0   
897                                          Certified          0      0.0   
898                                      Non-Certified          0      0.0   
899                                          Certified          0      0.0   
900                                      Non-Certified          0      0.0   
901                            Certified/Non-Certified          0      0.0   
902                                      Non-Certified          0    

In [42]:
with_codes[(with_codes["district"].str.contains("Westport"))\
          & with_codes["school_code"].isnull()]

district                school  \
11653  Westport School District  Greens Farms Academy   

                                           category       type 2010-11  \
11653  General Education - Teachers and Instructors  Certified       0   

       2011-12  2012-13  2013-14  2014-15 school_code check_sch_col  \
11653      0.0      0.0      0.0      1.0        None          None   

      check_dist_col  
11653           None

In [46]:
for sch in with_codes[ with_codes["school_code"].isnull()]["school"].unique():
    print sch

Wintergreen Interdistrict Magnet School
J. P. Vincent School
Central at Night
New Start Program
The University School
#NAME?
Greater Hartford Academy of the Arts High School - Full Time
Medical Professions and Teacher Preparation Academy
Academy of Aerospace & Engineering - Site 2
Charter School for Young Children on Asylum Hill Inc.
C. E. S. Special Education
Coventry Early Intervention Services
Mill Ridge Educational Center
Other
Willowbrook Early Childhood
Saint Christopher School
Arts at the Capitol Theater
GFLC/ACCESS School
Harriet Beecher Stowe School
Thomas G. Alcorn School
Asian Studies Academy
Breakthrough II Elementary School
Dr. Ramon E. Betances Early Reading Lab
Hartford Transitional Learning Academy (Secondary)
High School Inc.
Montessori Magnet School at Annie Fisher
R.J. Kinsella Magnet School of Performing Arts
University High of Science and Engineering
Weaver High School
Great Path Academy High School at MCC
Chalk Hill School
Jockey Hollow STEM Program
Hillside Middl

In [48]:
map(lambda x: x["school"], lookup_contains("district","Monroe"))

[u'Stepney Elementary School',
 u'REAL',
 u'Monroe School District',
 u'Monroe Elementary School',
 u'Masuk High School',
 u'Masuk Alternative Program',
 u'Jockey Hollow School',
 u'Fawn Hollow Elementary School']

[u'Bethel School District',
 u'Bethel Middle School',
 u'Bethel High School',
 u'Saint Mary School-Bethel']

# Good enough for now. What are the teaching categories?


In [54]:
with_codes["category"].value_counts()

Special Education - Teachers and Instructors                                     1236
General Education - Teachers and Instructors                                     1197
Counselors  Social Workers  and School Psychologists                             1176
Administrators  Coordinators and Department Chairs - School Level                1164
Other Staff Providing Non-Instructional Services/Support                         1094
General Education - Paraprofessional Instructional Assistants                    1090
School Nurses                                                                    1090
Special Education - Paraprofessional Instructional Assistants                    1090
Library/Media - Support Staff                                                    1090
Instructional Specialists Who Support Teachers                                   1078
Library/Media - Specialists (Certified)                                           869
 Research and Zoological Studies HS at the Fairchild-W

In [ ]:
with_codes["category"].value_counts()

In [55]:
with_codes["type"].value_counts()

Certified                                                            6720
Non-Certified                                                        4360
Certified/Non-Certified                                              1094
Administrators  Coordinators and Department Chairs - School Level       3
0                                                                       2
Name: type, dtype: int64

In [56]:
with_codes.to_csv("output/fte-levels.csv")

In [57]:
with_codes.to_csv("../clean/csv/fte-levels.csv")